xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-watch/src/main/java/org/xwiki/notifications/filters/watch/script/NotificationWatchScriptService.java

```diff
@@ -117,7 +118,30 @@ public WatchedEntityReference.WatchedStatus getLocationWatchedStatus(EntityRefer
             throws NotificationException
     {
         return watchedEntityFactory.createWatchedLocationReference(location)
-                .getWatchedStatus(documentAccessBridge.getCurrentUserReference());
+                .getWatchedStatus(CurrentUserReference.INSTANCE);
+    }
+
+    /**
+     * Try to retrieve the first ancestor of the location which have a status whose status is watched or blocked.
+     * This method returns {@code null} if current watch status is already watched or blocked, and if
+     * no ancestor can be found. The first matching ancestor reference is returned along with its computed
+     * {@link WatchedEntityReference.WatchedStatus}.
+     * @param location the location for which to find an ancestor with a watched status
+     * @return a pair containing the watched entity reference of the matching ancestor and its watched status or
+     * {@code null}.
+     * @see WatchedEntityReference.WatchedStatus#isWatched()
+     * @see WatchedEntityReference.WatchedStatus#isBlocked()
+     * @throws NotificationException in case of problem for computing the status
+     * @since 16.4.0RC1
```

Comment:
Wrong since ?

Reply:
Right... sounds like I forgot to update them. I'll make a pass over all of them again. Sorry about that.